### PR TITLE
Create RSpec custom matchers

### LINKED ISSUE #51: Custom matchers for RSpec?
Hey @hzamani,

first of all, thank you very much for this gem. It already helped me a lot on past projects, and it's helping me again :)

On a current project, I noticed that I wanted to do a quick check on classes and instances of those classes to see if they were acting as/actables of a certain class or symbol, so I dig up your specs and saw that you have some nice boolean methods (`#acting_as?` + `.acting_as?` + `.actable?`) to verify that. Do you think it's interesting adding some custom matchers to the library to help people out writing specs?

Example:

```ruby
# Sample rails_helper.rb

require 'rspec/acts_as_matchers'

# acts_as_matchers.rb

RSpec::Matchers.define :acts_as do |actable|
  match { |actor| actor.acting_as?(actable) }
end

RSpec::Matchers.define :be_actable do
  match { |instance| instance.class.actable? }
end
```

And then on specs, people would have available the following interfaces:

```ruby
it { is_expected.to acts_as(:account) }
it { expect(Person).to acts_as(:account) }

it { is_expected.to acts_as(Account) }
it { expect(Person).to acts_as(Account) }

it { expect(described_class).to be_actable }
```

What do you think? :)

If you like the idea, I can open a PR with those changes right away. If not, I'll just close the issue.

### PR DESCRIPTION
**Pull request by tiagoamaro**
closes #51 

This commit also introduces documentation in the README file to explain how to use the custom matchers

### CODE REVIEW

File: lib/rspec/acts_as_matchers.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+RSpec::Matchers.define :acts_as do |actable|
```
Comment (lowjoel) [inline lib/rspec/acts_as_matchers.rb]: I'd vote for keeping this singular.

Comment (tiagoamaro) [inline lib/rspec/acts_as_matchers.rb]: You're right! I was thinking like "an object that acts as", but I think the proper phrasing would be "an object that would act as" :)

**Comment by hzamani** [comment]
No test?

**Comment by hzamani** [comment]
:-)

**Pull request merged**